subsys/usb/usb_c/usbc_pe_snk_states.c

```diff
@@ -607,12 +612,40 @@ void pe_snk_get_source_cap_run(void *obj)
 	const struct device *dev = pe->dev;
 	struct usbc_port_data *data = dev->data;
 	struct protocol_layer_rx_t *prl_rx = data->prl_rx;
+	union pd_header header;
 
 	/* Wait until message is sent or dropped */
 	if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_TX_COMPLETE)) {
+		/* The Policy Engine Shall then start the SenderResponseTimer. */
+		usbc_timer_start(&pe->pd_t_sender_response);
+	}
+	/*
+	 * The Policy Engine Shall transition to the PE_SNK_Evaluate_Capability
+	 * State when:
+	 *	1: In SPR Mode and SPR Source Capabilities were requested and
+	 *	   a Source_Capabilities Message is received
+	 */
+	else if (atomic_test_and_clear_bit(pe->flags, PE_FLAGS_MSG_RECEIVED)) {
+		header = prl_rx->emsg.header;
+
+		if (received_control_message(dev, header, PD_DATA_SOURCE_CAP)) {
+			pe_set_state(dev, PE_SNK_EVALUATE_CAPABILITY);
+			/*
+			 * Inform the DPM of the reception of the source
+			 * capabilities
+			 */
+			policy_notify(dev, SOURCE_CAPABILITIES_RECEIVED);
```

Comment:
It seems more foolproof to just do this unconditionally in `pe_snk_evaluate_capability_entry`.

Also, the arrangement here breaks the "return immediately after `pe_set_state` rule. I don't think that would actually be a problem here, but it's probably easier if we keep it absolute.

Reply:
I didn't want to send the notification until the Source Caps message was received and I switched the order of the notification and state change.

Reply:
I'm OK with this as is, but wouldn't calling `policy_notify` from `pe_snk_evaluate_capability_entry` still guarantee that the notification doesn't send until the Source Caps have been received?

Reply:
I guess I needed to read your suggestion twice. I've moved the notification to pe_snk_evaluate_capability_entry.